lib/commands/log.js

```diff
@@ -37,4 +41,66 @@ extensions.startLogCapture = async function () {
   return true;
 };
 
+/**
+ * Starts iOS system logs broadcast websocket on the same host and port
+ * where Appium server is running at `/ws/session/:sessionId:/appium/syslog` endpoint. The method
+ * will return immediately if the web socket is already listening.
+ *
+ * Each connected webcoket listener will receive syslog lines
+ * as soon as they are visible to Appium.
+ */
+extensions.mobileStartLogsBroadcast = async function () {
+  const pathname = WEBSOCKET_ENDPOINT(this.sessionId);
+  if (!_.isEmpty(await this.server.getWebSocketHandlers(pathname))) {
+    log.debug(`The system logs broadcasting web socket server is already listening at ${pathname}`);
+    return;
+  }
+
+  log.info(`Assigning system logs broadcasting web socket server to ${pathname}`);
+  // https://github.com/websockets/ws/blob/master/doc/ws.md
+  const wss = new WebSocket.Server({
+    noServer: true,
+  });
+  wss.on('connection', (ws, req) => {
+    if (req) {
+      const remoteIp = _.isEmpty(req.headers['x-forwarded-for'])
+        ? req.connection.remoteAddress
+        : req.headers['x-forwarded-for'];
+      log.debug(`Established a new system logs listener web socket connection from ${remoteIp}`);
+    } else {
+      log.debug('Established a new system logs listener web socket connection');
+    }
+    this.logs.syslog.on('output', (logRecord) => {
```

Comment:
do we want to remove this event listener whenever we stop the logs broadcast?

Reply:
good catch. This has to be fixes for Android as well